src/quo/components/tabs/tabs/view.cljs

```diff
@@ -118,87 +119,99 @@
-             ;; The padding-top workaround is needed because on Android
-             ;; {:overflow :visible} doesn't work on components inheriting
-             ;; from ScrollView (e.g. FlatList). There are open issues, here's
-             ;; just one about this topic:
-             ;; https://github.com/facebook/react-native/issues/31218
-             :content-container-style
-             (assoc container-style :padding-top (dec unread-count-offset))
-             :horizontal true
-             :scroll-event-throttle 64
-             :shows-horizontal-scroll-indicator false
-             :data data
-             :key-fn (comp str :id)
-             :on-scroll-to-index-failed identity
-             :on-scroll (partial on-scroll-handler
-                                 {:fade-end-percentage fade-end-percentage
-                                  :fade-end?           fade-end?
-                                  :fading              fading
-                                  :on-scroll           on-scroll})
-             :render-fn tab-view
-             :render-data {:active-tab-id       active-tab-id
-                           :blur?               blur?
-                           :customization-color customization-color
-                           :flat-list-ref       flat-list-ref
-                           :number-of-items     (count data)
-                           :on-change           on-change
-                           :scroll-on-press?    scroll-on-press?
-                           :size                size
-                           :style               style}})]]]
-        [rn/view (merge style {:flex-direction :row})
-         (map-indexed (fn [index item]
-                        ^{:key (:id item)}
-                        [tab-view item index nil
-                         {:active-tab-id       active-tab-id
-                          :blur?               blur?
-                          :customization-color customization-color
-                          :number-of-items     (count data)
-                          :on-change           on-change
-                          :size                size
-                          :style               style}])
-                      data)]))))
+  [{:keys [default-active
+           data
+           fade-end-percentage
+           fade-end?
+           on-change
+           on-scroll
+           scroll-on-press?
+           scrollable?
+           style
+           container-style
+           size
+           blur?
+           in-scroll-view?
+           customization-color]
+    :or   {fade-end-percentage 0.8
+           fade-end?           false
+           scrollable?         false
+           scroll-on-press?    false
+           size                default-tab-size}
+    :as   props}]
+  (let [[active-tab-id set-active-tab-id] (rn/use-state default-active)
+        [fading set-fading]               (rn/use-state {:fade-end-percentage fade-end-percentage})
```

Comment:
I originally coded a Reagent atom as a map because I wasn't sure if the fading calculation would change in the future to be more customizable. There was some uncertainty from designers. I see now we don't need to have a map as the state.